Improve high DPI support in wxAui

rozmansi opened the pull request.

rozmansi committed:
Add VS2017 solution and project to auidemo sample

rozmansi committed:
Improve high DPI support in wxAui

Window initial/minimum/maximum sizes are now treated as logical pixels.

Furthermore, many margins and paddings are now converted using
wxWindow::FromDIP() to allow their growth in accord with screen DPI.
This places buttons on toolbars more apart on high DPI screens providing
space for easier touch operations.

vadz commented:
Thank you, I'm applying the second (significant) commit, but not the first one as I think that we should either have MSVS 2017 solutions for all samples or for none of them (and as we unfortunately still don't have any reasonable way of maintaining them for all of them, for now it will be none).

rozmansi commented:
I have introduced VS2017 sln/vcxproj to test all my modifications live. Indeed, it has no added value for the rest of the world.

I have also added some `// TODO` comments regarding the 16×16px bitmap artworks. We'll need to provide 24×24px and 32×32px variants for better experience. Or, switch to SVG/GDI artworks. Grippers are already using GDI painting and with this PR they scale nicely now.
